<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>GUI_ParamDeform</class>
 <widget class="QMainWindow" name="GUI_ParamDeform">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>600</width>
    <height>400</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>GUI_ParamDeform</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab_4">
       <attribute name="title">
        <string>Tab 1</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_5">
        <item row="0" column="0" rowspan="6">
         <widget class="PhotoView" name="viewPic"/>
        </item>
        <item row="0" column="1" colspan="2">
         <widget class="QTabWidget" name="tabProcessMethod">
          <property name="currentIndex">
           <number>0</number>
          </property>
          <widget class="QWidget" name="tab_3">
           <attribute name="title">
            <string>FreeDeform</string>
           </attribute>
           <layout class="QGridLayout" name="gridLayout_3">
            <item row="0" column="0">
             <widget class="Widget_WarpParam" name="wWarpParam" native="true"/>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="tab_2">
           <attribute name="title">
            <string>Param</string>
           </attribute>
           <layout class="QGridLayout" name="gridLayout_2">
            <item row="0" column="0">
             <widget class="QSlider" name="sldFatThin">
              <property name="minimum">
               <number>-50</number>
              </property>
              <property name="maximum">
               <number>50</number>
              </property>
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="tab">
           <attribute name="title">
            <string>RawParam</string>
           </attribute>
           <layout class="QGridLayout" name="gridLayout_4">
            <item row="0" column="0">
             <widget class="Widget_ModelParamEdit" name="wParamEdit" native="true"/>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Image Warping </string>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QComboBox" name="cmbImgWarpAlg"/>
        </item>
        <item row="2" column="2">
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>219</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="3" column="1" colspan="2">
         <widget class="QCheckBox" name="chkShowMarkPoint">
          <property name="text">
           <string>Show Mark Points</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="4" column="1">
         <widget class="QCheckBox" name="chkShowImage">
          <property name="text">
           <string>Show Image</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="5" column="1">
         <widget class="QCheckBox" name="chkDeform">
          <property name="text">
           <string>Deform</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_5">
       <attribute name="title">
        <string>Tab 2</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_6">
        <item row="0" column="0">
         <widget class="Widget_FaceMorph" name="widget" native="true"/>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>600</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionLoadImg"/>
    <addaction name="actCapFromCam"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
    <addaction name="actionZoom_In"/>
    <addaction name="actionZoom_out"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuView"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionLoadImg">
   <property name="text">
    <string>Load Image...</string>
   </property>
  </action>
  <action name="actCapFromCam">
   <property name="text">
    <string>Capture from camera...</string>
   </property>
  </action>
  <action name="actionZoom_In">
   <property name="text">
    <string>Zoom In</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+=</string>
   </property>
  </action>
  <action name="actionZoom_out">
   <property name="text">
    <string>Zoom Out</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+-</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>PhotoView</class>
   <extends>QGraphicsView</extends>
   <header>photoview.h</header>
   <slots>
    <signal>selectedPointUpdated(int)</signal>
    <signal>warpControlUpdated()</signal>
    <slot>setMirrorPoint(int)</slot>
    <slot>setImageVisibility(bool)</slot>
    <slot>setMarkPointVisibility(bool)</slot>
   </slots>
  </customwidget>
  <customwidget>
   <class>Widget_ModelParamEdit</class>
   <extends>QWidget</extends>
   <header>widget_modelparamedit.h</header>
   <container>1</container>
   <slots>
    <signal>paramUpdated()</signal>
   </slots>
  </customwidget>
  <customwidget>
   <class>Widget_WarpParam</class>
   <extends>QWidget</extends>
   <header>widget_warpparam.h</header>
   <container>1</container>
   <slots>
    <signal>updated()</signal>
   </slots>
  </customwidget>
  <customwidget>
   <class>Widget_FaceMorph</class>
   <extends>QWidget</extends>
   <header>widget_facemorph.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>chkShowMarkPoint</sender>
   <signal>toggled(bool)</signal>
   <receiver>viewPic</receiver>
   <slot>setMarkPointVisibility(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>486</x>
     <y>300</y>
    </hint>
    <hint type="destinationlabel">
     <x>206</x>
     <y>209</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>chkShowImage</sender>
   <signal>toggled(bool)</signal>
   <receiver>viewPic</receiver>
   <slot>setImageVisibility(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>485</x>
     <y>328</y>
    </hint>
    <hint type="destinationlabel">
     <x>226</x>
     <y>272</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tabProcessMethod</sender>
   <signal>currentChanged(int)</signal>
   <receiver>GUI_ParamDeform</receiver>
   <slot>updatePic()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>406</x>
     <y>55</y>
    </hint>
    <hint type="destinationlabel">
     <x>386</x>
     <y>40</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmbImgWarpAlg</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>GUI_ParamDeform</receiver>
   <slot>updatePic()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>595</x>
     <y>132</y>
    </hint>
    <hint type="destinationlabel">
     <x>473</x>
     <y>155</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>wWarpParam</sender>
   <signal>updated()</signal>
   <receiver>GUI_ParamDeform</receiver>
   <slot>updatePic()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>487</x>
     <y>89</y>
    </hint>
    <hint type="destinationlabel">
     <x>256</x>
     <y>39</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sldFatThin</sender>
   <signal>valueChanged(int)</signal>
   <receiver>GUI_ParamDeform</receiver>
   <slot>updatePic()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>487</x>
     <y>89</y>
    </hint>
    <hint type="destinationlabel">
     <x>521</x>
     <y>213</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>wParamEdit</sender>
   <signal>paramUpdated()</signal>
   <receiver>GUI_ParamDeform</receiver>
   <slot>updatePic()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>587</x>
     <y>93</y>
    </hint>
    <hint type="destinationlabel">
     <x>524</x>
     <y>200</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>viewPic</sender>
   <signal>warpControlUpdated()</signal>
   <receiver>GUI_ParamDeform</receiver>
   <slot>updatePic()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>307</x>
     <y>72</y>
    </hint>
    <hint type="destinationlabel">
     <x>272</x>
     <y>44</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>updatePic()</slot>
 </slots>
</ui>
